### PR TITLE
Add connection pooling limitation for Autoscaling

### PR DESCRIPTION
**Pull request by danieltprice**
Previews:
https://neon-next-git-dprice-autoscaling-connection-limit-neondatabase.vercel.app/docs/connect/connection-pooling
https://neon-next-git-dprice-autoscaling-connection-limit-neondatabase.vercel.app/docs/manage/endpoints#compute-size-and-autoscaling-configuration

### CODE REVIEW

File: content/docs/connect/connection-pooling.md
Comment (stepashka) [inline content/docs/connect/connection-pooling.md:11]: This is minor, but perhaps we can shorten the
'Autoscaling feature, which is currently in Beta' to 'Autoscaling (Beta)' or something along these lines?
would be great to be able to annotate with the Beta labels directly in the docs.

**Review by m-abdelwahab — APPROVED** [review]
LGTM. I'd wait for others to review before merging

**Review by stepashka — APPROVED** [review]

**Review by m-abdelwahab — APPROVED** [review]

**Pull request merged**
